docs/managing-vendor-dependencies.md

```diff
@@ -0,0 +1,176 @@
+
+To build the javascript file/s into the `vendor.js` dist file, we need to add all the javascript files to the [`vendor-core/lib/modules.js`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/lib/modules.js) file.
+
+In our case with `@patternfly/react-table`, there is only one javascript file so we should just add `@patternfly/react-table` to the exported array in the [`vendor-core/lib/modules.js`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/lib/modules.js) file:
+
+```diff
+'react',
+'react-dom',
+'@patternfly/react-core',
++'@patternfly/react-table',
+'redux',
+```
+
+### Step 3: Provides s/css with the package
+
+In some cases, a package also provides css or scss files.
+To have the css available to the vendor consumers (foreman and it's plugins), we should add them to the [`vendor-core/scss/vendor-core.scss`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/scss/vendor-core.scss) so they would get compiled into the `vendor.css` dist file.
+
+When a package provides variables and mixins, we should add them to [`vendor-core/scss/variables.scss`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/scss/variables.scss) and [`vendor-core/scss/mixins.scss`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/scss/mixins.scss) files.
+They would get compiled into `@theforeman/vendor/scss/variables.scss` and `@theforeman/vendor/scss/mixins.scss` files.
+The consumers can import those files and reuse the package variables or mixins files.
+
+**In our cases with the `@patternfly/react-table`, there are no s/css files so we can skip this step.**
+
+### Step 4: Build, and test with the foreman or with a foreman plugin
+
+Once we have the package installed, listed in the `modules.js` file and the `.scss` files, we should be able to build and use the package from foreman or a foreman plugin.
+
+Build:
+```sh
+npm run build
+```
+
+Link the local vendor version to the foreman or a foreman plugin:
+```sh
+npm run link -- --location ../foreman
+# or for a plugin
+npm run link -- --location ../foreman-tasks
+```
+
+Now you should use the newly added package in foreman or a foreman plugin (foreman-tasks in our case).
+Just import and use it as you would typically do with an npm package and run foreman.
+
+### Step-5: commit and create a pull-request
+
+The foreman-js project uses CI automation to read the commit-message and automatically publish a new version to npm.
+
+To create a valid commit-message, we should use the `npm run commit` command provided by the `foreman-js` project:
+```sh
+git add .
+npm run commit
+```
+
+You will need to answer for a set of questions, and the script should create a commit message for you.
+
+When answering the questions:
+- **"Select the type of change that you're committing"**
+  - always choose **" feat: A new feature"** since we are adding a new feature to the vendor
```

Comment:
```suggestion
  - always choose **" feat: A new feature"** since we are adding a new feature to the vendor.
```

It gets rendered on a single line and produces `...  to the vendor The automation process ...` which looks odd